src/main/com/fulcrologic/fulcro/algorithms/normalized_state_helpers.cljc

```diff
@@ -0,0 +1,491 @@
+   to the mutation caller (which can be explicitly set via `:ref` when calling `transact!`).
+
+   Equivalent to `(swap! (:state env) update-in (:ref env) ...)`."
+  [{:keys [state ref] :as mutation-env} & args]
+  (apply swap! state update-in ref args))
+
+
+(comment
+
+
+  (let [mutation-env {:ref   [:person/id 1]
+                      :state (atom {:person/id {1
+                                                {:person/id 1 :person/name "Dad"}}})}]
+    (apply swap! (:state mutation-env) update-in (:ref mutation-env)
+           (vector assoc :person/name "Mom")))
+
+
+
+  (let [mutation-env {:ref   [:person/id 1]
+                      :state (atom {:person/id {1
+                                                {:person/id 1 :person/name "Dad"}}})}]
+    (update-caller! mutation-env
+                    assoc :person/name "Mom"))
+
+
+
+
+  '())
+
+;============================================================================
+
+;;THOUGHT These might belong in mutation ns???
+(defn update-caller-in!
+  "Like swap! but starts at the ref from `env`, adds in supplied `path` elements
+  (resolving across idents if necessary). Finally runs an update-in on that resultant
+  path with the given `args`.
+
+   Roughly equivalent to:
+
+   ```
+   (swap! (:state env) update-in (tree-path->db-path @state (into (:ref env) path)) args)
+   ```
+
+   with a small bit of additional sanity checking."
+  [{:keys [state ref] :as mutation-env} path & args]
+  (let [path (tree-path->db-path @state (into ref path))]
+    (if (and path (get-in @state path))
+      (apply swap! state update-in path args)
+      @state)))
+
+
+
+(comment
+
+  (let [state (atom {:person/id {1 {:person/id 1 :person/name "Dad" :person/children [[:person/id 2] [:person/id 3]]}
+                                 2 {:person/id 2 :person/name "Son"}
+                                 3 {:person/id 3 :person/name "Daughter"}}})
+        ref [:person/id 1]
+        path (tree-path->db-path @state (into ref [:person/id 2]))
+        args (vector assoc :person/name "Mom")]
+
+    (if (and path (get-in @state path))
+      (apply swap! state update-in path args)
+      @state))
+
+
+
+
+
+  (let [state (atom {:person/id {1 {:person/id 1 :person/name "Dad" :person/children [[:person/id 2] [:person/id 3]]}
+                                 2 {:person/id 2 :person/name "Son"}
+                                 3 {:person/id 3 :person/name "Daughter"}}})
+        ref [:person/id 1]
+        path (tree-path->db-path @state (into ref [:person/id 2]))
+        args (vector assoc :person/name "Mom")]
+
+    (and path (get-in @state path)))
+
+
+
+
+  ;;;;;
+
+
+  '())
+
+;============================================================================
+
+;; TODO: Untested...make up an env with a state atom and see if it works in clj/cljs
+
+#?(:clj
+   (defmacro swap*                                          ;; or alternate name could be `state-swap!`
```

Comment:
Where did the arrow in the name go???  This should be `swap*->` or `swap!->` or something of the sort...threading is definitely implied.

Reply:
Oh, I thought it was more of a scratch name since you were looking for a different option. No worries, I'll add it back.